kernel/kernel-impl/src/main/webapp/WEB-INF/grouplock-components.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:tx="http://www.springframework.org/schema/tx"
+       xsi:schemaLocation="
+        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd
+        http://www.springframework.org/schema/tx http://www.springframework.org/schema/tx/spring-tx.xsd">
+
+    <context:annotation-config/>
+    <tx:annotation-driven transaction-manager="org.sakaiproject.springframework.orm.hibernate.GlobalTransactionManager"/>
+
+    <bean id="sakai.site.mappings"
+       class="org.sakaiproject.springframework.orm.hibernate.impl.AdditionalHibernateMappingsImpl">
+       <property name="annotatedClasses">
+          <list>
+             <value>org.sakaiproject.site.api.GroupLock</value>
+          </list>
+         </property>
+    </bean>
+
+    <!-- class="org.sakaiproject.messagebundle.impl.CachingMessageBundleServiceImpl"-->
+	<bean id="org.sakaiproject.site.api.GroupLockService"
+          class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
+        <property name="target">
+            <bean class="org.sakaiproject.site.impl.GroupLockServiceImpl">
+                <property name="sessionFactory" ref="org.sakaiproject.springframework.orm.hibernate.GlobalSessionFactory"/>
+            </bean>
+        </property>
+        <property name="transactionManager" ref="org.sakaiproject.springframework.orm.hibernate.GlobalTransactionManager"/>
```

Comment:
this isn't need if your using tx:annotation-driven and marking the service with methods with @Transactional

Reply:
I tried removing it, but it was causing errors. So I decided to remove the tx:annotation-driven tag.